lib/semantics/check-omp-structure.h

```diff
@@ -147,6 +149,25 @@ class OmpStructureChecker : public virtual BaseChecker {
     return nullptr;
   }
 
+  template<typename T> std::optional<std::int64_t> GetIntValue(const T &x) {
+    const auto *expr{GetExpr(x)};
+    if (expr && evaluate::ToInt64(*expr).has_value()) {
+      return evaluate::ToInt64(*expr).value();
+    }
+    return std::nullopt;
+  }
+  template<typename T> std::int64_t GetPosIntValue(const T &x) {
+    const auto &v{GetIntValue(x)};
+    return v && v > 0 ? v.value() : 0;
+  }
+  template<typename T> std::int64_t GetConstPosIntValue(const T &x) {
+    const auto *expr{GetExpr(x)};
+    if (expr && IsConstantExpr(*expr)) {
```

Comment:
Make sure that the definition of "constant expression" used by `IsConstantExpr()` is indeed what you want.

```
if (const auto *expr{GetExpr(x)}) {
  if (IsConstantExpr(*expr)) {
    return ...
  }
}
return 0;
```

Also, you had an explicit `evaluate::` namespace on the call to `ToInt64` earlier, but you don't have one here on the call to `IsConstantExpr()`.  Please be consistent if possible.

Reply:
Addressed in commit [8aa3737](https://github.com/flang-compiler/f18/pull/540/commits/8aa373705b1f740ddf49345affc5a1fa0bbd84d6). After talking with Tim, this `GetConstPosIntValue` function is removed.